### PR TITLE
Integrate questions endpoint and improve UX

### PR DESCRIPTION
**Pull request by BlakeDonn**
#### What's this PR do?  

- This PR removes a few bugs regarding user interaction 
- User was allowed to select as many answers as possible before, they are now limited
- User would be presented with the `how many players do you have` option twice if both games were selected
- Adds initial `submit` handling for when a user has selected answers to all the questions

#### Where should the reviewer start? 

- Check state on all parts of the user interaction process to ensure functionality is correct

#### How should this be manually tested?  

- Try and break this through edge cases and other weird interactions 
- Go through user stories as randomly as possible to test all avenues

#### Any background context you want to provide?  

- I believe all the logic is solid here, I have yet to find any bugs that have arisen from these changes

#### What are the relevant tickets?  
 
#39 
#45  
#46
#47

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
+import {CardGame} from '../CardGame/CardGame';
 import React, {Component} from 'react';
-import {Route} from 'react-router-dom';
-import { getAllMovies, getAllPodcasts, getAllCardGames, getAllMusic, getAllBoardGames} from '../../apiCalls.js'
-import { BrowsePage } from '../BrowsePage/BrowsePage'
-import { Footer } from '../Footer/Footer'
+import {Route, Switch} from 'react-router-dom';
+import {getAllMovies, getAllPodcasts, getAllCardGames, getAllMusic, getAllBoardGames} from '../../apiCalls.js'
+import {BrowsePage} from '../BrowsePage/BrowsePage'
+import {Footer} from '../Footer/Footer'
```
Comment (Atos20) [inline src/Components/App/App.js]: I noticed that from lines 3 to 11, you removed the spaces in between the curly braces. My understanding is that we want those spaces for better readability.

File: src/Components/App/App.js [hunk 2]
```diff
@@ -32,86 +32,104 @@ class App extends Component {
...
     }
   }
 
   setActivities = (activities) => {
     this.setState({activities: [...activities]})
   }
 
-  render() {
+  determineRandomActivity = () => {
+    console.log('yeet')
```
Comment (nicolegooden) [inline src/Components/App/App.js:73]: Let's get this `console log` out of here!

File: src/Components/Form/Form.js [hunk 3]
```diff
@@ -1,91 +1,119 @@
...
+    if (+questionData.allowedResponses && this.state.currentAnswers.find(answer => questionData.choices.includes(answer))) {
+      let choiceToSwap = this.state.currentAnswers.find(answer => questionData.choices.includes(answer))
+      let indexToReplace = this.state.currentAnswers.indexOf(choiceToSwap)
+      let answerToReplace = this.state.currentAnswers.slice(indexToReplace + 1)
+      if (answerToReplace) {
+        return this.setState({currentAnswers: [...answerToReplace, event.target.textContent]})
+      }
+    }
+    else {
+      this.setState({currentAnswers: [...this.state.currentAnswers, event.target.textContent]})
```
Comment (Atos20) [inline src/Components/Form/Form.js]: Thanks for adding this piece of logic and it is easy to read as well.

File: src/Components/App/App.js [hunk 4]
```diff
@@ -32,86 +32,104 @@ class App extends Component {
...
+    // redirect to temporary loading page while the below logic is run
+    // what activities did the user select?
+    // fetch all activities and place somewhere to be filtered through, here? Result component?
```
Comment (nicolegooden) [inline src/Components/App/App.js:76]: I believe the fetch is happening in `App.js` so we'd only need to pass the appropriate data as props.  I think maybe we should keep the filtering in `App.js` so that we could keep `Result` as a functional component.

File: src/Components/App/App.js [hunk 5]
```diff
@@ -32,86 +32,104 @@ class App extends Component {
...
+    // if Result component we will also have to pass the answers of the questions down
+    // filter through All results from fetch with specific conditions based on selected answers
+    // store these further filtered results in somewhere where we can access them more than once, state?
+    // randomly choose one of these further filtered activities, maybe remove this from state at that time? 
+    // render component that will display the previewcard
+    // let user select to see more info or skip
+    // if user selects skip, get another random activity from set which should no longer include the one they skipped if we removed it
```
Comment (nicolegooden) [inline src/Components/App/App.js:83]: I like your thought process here about removing the skipped activity from state; would lead to better use experience in the event that skipped activity were to show up twice.

**Review by Atos20 — APPROVED** [review]
Thank you for handling the form logic, this is a great addition to the project.

File: src/Components/App/App.js [hunk 6]
```diff
@@ -32,86 +32,104 @@ class App extends Component {
...
+    // Should user be able to skip once they have selected to see more info? If they are no longer interested?
+  }
```
Comment (nicolegooden) [inline src/Components/App/App.js:85]: Yes, I think that would be a good thing to implement for the user.  This means that the user would need to be able to go back to the `Result` component, where maybe a new suggestion should be rendered upon load?

File: src/Components/Form/Form.js [hunk 7]
```diff
@@ -1,91 +1,119 @@
...
+      }
+      relevantQuestions.push(questionsByActivity)
+      return relevantQuestions;
+    }, [])
+  }
+
+  handleBothGames = () => {
+    if (this.state.allAnswers[0].includes('card games') && this.state.allAnswers[0].includes('board games')) {
+      let filteredSet = this.state.allAnswers[0].sort((a, b) => {
+        return a > b ? - 1 : 1
```
Comment (nicolegooden) [inline src/Components/Form/Form.js:62]: It was interesting to talk through what's happening on line 62.  Will keep in mind that this syntax is helpful when trying to sort letters rather than numbers.

File: src/apiCalls.js [hunk 8]
```diff
@@ -61,4 +61,13 @@ export const getPodcastsByGenre = async (genre) => {
   } else {
     return response.error
   }
+
+}
+export const getAllQuestions = async () => {
+  const response = await fetch('http://localhost:3000/api/v1/questions')
+  if (response.ok) {
```
Comment (nicolegooden) [inline src/apiCalls.js:69]: Nice error handling here.

**Review by nicolegooden — APPROVED** [review]

File: src/Components/App/App.js (hunk 1 above)
Comment (nicolegooden) [reply src/Components/App/App.js]: @Atos20  I thought so too, it's interesting that the linter doesn't like it.

Comment (BlakeDonn) [reply src/Components/App/App.js]: Yeah I need to adjust my linter, it defaults to remove the spaces here. Sorry about that
